sgkit/api.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Hashable, List, Mapping
+
+import xarray as xr
+
+from .utils import check_array_like
+
+DIM_VARIANT = "variants"
+DIM_SAMPLE = "samples"
+DIM_PLOIDY = "ploidy"
+DIM_ALLELE = "alleles"
+DIM_GENOTYPE = "genotypes"
+
+
+def create_genotype_call_dataset(
+    variant_contig_names: List[str],
+    variant_contig: Any,
+    variant_position: Any,
+    variant_alleles: Any,
+    sample_id: Any,
+    call_genotype: Any,
+    call_genotype_phased: Any = None,
+    variant_id: Any = None,
+) -> xr.Dataset:
+    """Create a dataset of genotype calls.
+
+    Parameters
+    ----------
+    variant_contig_names : list of str
+        The contig names.
+    variant_contig : array_like, int
+        The (index of the) contig for each variant.
+    variant_position : array_like, int
+        The reference position of the variant.
+    variant_alleles : array_like, S1
+        The possible alleles for the variant.
+    sample_id : array_like, str
+        The unique identifier of the sample.
+    call_genotype : array_like, int
+        Genotype, encoded as allele values (0 for the reference, 1 for
+        the first allele, 2 for the second allele), or -1 to indicate a
+        missing value.
+    call_genotype_phased : array_like, bool, optional
+        A flag for each call indicating if it is phased or not. If
+        omitted all calls are unphased.
+    variant_id: array_like, str, optional
+        The unique identifier of the variant.
+
+    Returns
+    -------
+    xr.Dataset
+        The dataset of genotype calls.
+
+    """
+    check_array_like(variant_contig, kind="i", ndim=1)
+    check_array_like(variant_position, kind="i", ndim=1)
+    check_array_like(variant_alleles, kind="S", ndim=2)
+    check_array_like(sample_id, kind="U", ndim=1)
+    check_array_like(call_genotype, kind="i", ndim=3)
+    data_vars: Mapping[Hashable, Any] = {
+        "variant/contig": ([DIM_VARIANT], variant_contig),
```

Comment:
Shouldn't this be 
```python
f"{DIM_VARIANT}/contig": (...)
```
etc?

Reply:
I think there is (or can be) a distinction between the prefixes that we use for the different arrays, and the names of the dimensions.

E.g.,, we could have arrays named like "variant/position", "variant/id", "sample/id", "call/GT", etc.

And we could have dimension names like "variants", "samples", "ploidy", etc.

There is an implicit convention here that:

* arrays with the prefix "variant/" can be 1D or 2D but all have dimensions (variants,) or (variants, foo) where foo is some other dimension than samples
* arrays with the prefix "sample/" are all 1D and have dimensions (samples,) 
* arrays with the prefix "call/" are all 2D or 3D and have dimensions (variants, samples) or (variants, samples, bar) where bar can be ploidy or something else.

Reply:
OK, I'm happy to go with whatever the usual conventions are.